File: internal/routers/nodes.go
package routers

import (
	"github.com/fluent-manager/fluent-manager/internal/middleware"
	"github.com/gin-gonic/gin"
)

func registerNodeRoutes(authed *gin.RouterGroup, h *allHandlers) {
	nodes := authed.Group("/nodes")
	{
		nodes.GET("", middleware.RequirePermission("nodes", "read"), h.Node.List)
		nodes.GET("/stats", middleware.RequirePermission("nodes", "read"), h.Node.Stats)
		nodes.GET("/:id", middleware.RequirePermission("nodes", "read"), h.Node.Get)
		nodes.PUT("/:id", middleware.RequirePermission("nodes", "update"), h.Node.Update)
		nodes.DELETE("/:id", middleware.RequirePermission("nodes", "delete"), h.Node.Delete)
		nodes.POST("/batch-move", middleware.RequirePermission("nodes", "update"), h.Node.BatchMoveCluster)
		// Node metrics, logs, remote commands
		nodes.GET("/:id/metrics", middleware.RequirePermission("nodes", "read"), h.Agent.GetNodeMetrics)
		nodes.GET("/:id/logs", middleware.RequirePermission("nodes", "read"), h.Agent.GetNodeLogs)
		nodes.POST("/:id/commands", middleware.RequirePermission("nodes", "update"), h.Agent.SendCommand)
		nodes.GET("/:id/commands", middleware.RequirePermission("nodes", "read"), h.Agent.ListNodeCommands)
		nodes.GET("/:id/commands/:cmdID", middleware.RequirePermission("nodes", "read"), h.Agent.GetNodeCommand)
		nodes.GET("/:id/fluent-profile", middleware.RequirePermission("nodes", "read"), h.Fluent.GetNodeProfile)
		nodes.PUT("/:id/fluent-profile", middleware.RequirePermission("nodes", "update"), h.Fluent.UpdateNodeProfile)
	}
}

File: web/frontend/src/api/nodes.ts
import api from './client'

export const getNodes = (params?: Record<string, any>) => api.get('/nodes', { params })
export const getNode = (id: number) => api.get(`/nodes/${id}`)
export const updateNode = (id: number, data: Record<string, any>) => api.put(`/nodes/${id}`, data)
export const deleteNode = (id: number) => api.delete(`/nodes/${id}`)
export const getNodeStats = () => api.get('/nodes/stats')
export const batchMoveCluster = (data: Record<string, any>) => api.post('/nodes/batch-move', data)
export const getNodeMetrics = (id: number) => api.get(`/nodes/${id}/metrics`)
export const getNodeLogs = (id: number) => api.get(`/nodes/${id}/logs`)
export const sendNodeCommand = (id: number, data: Record<string, any>) => api.post(`/nodes/${id}/commands`, data)
export const getNodeCommands = (id: number) => api.get(`/nodes/${id}/commands`)
export const getNodeCommand = (id: number, cmdId: number) => api.get(`/nodes/${id}/commands/${cmdId}`)
